integrations/azure_ai_search/src/haystack_integrations/document_stores/azure_ai_search/document_store.py

```diff
@@ -426,16 +485,31 @@ def _convert_search_result_to_documents(self, azure_docs: List[Dict[str, Any]])
             if embedding == self._dummy_vector:
                 embedding = None
 
+            score = None
```

Comment:
```suggestion
            score = azure_doc.get("@search.score")
```

I would do this always, TBH. @Amnah199 agree?

Reply:
I think this doesn't work if the user has `include_search_metadata = False` right? Or do we want the behavior to be that `Document.score` is always populated now, no matter that flag?